### PR TITLE
Foregrounds, CMB, extragalactic simulations with tophat bandpasses

### PR DESCRIPTION
**Pull request by zonca**

### CODE REVIEW

**Comment by zonca** [comment]
@keskitalo I have run the SAT maps at NSIDE 512, now I need to create the combined components,

do you confirm these are the 2 we need?

```python
all_combined = {
    "combined_foregrounds": ["dust", "synchrotron", "freefree", "ame"]
    + ["cib", "ksz", "tsz"],
    "cmb_lensing_signal": ["cmb", "-cmb_unlensed"],
}
```

**Comment by zonca** [comment]
maps are at `/global/cscratch1/sd/zonca/cmbs4/map_based_simulations/202002_foregrounds_extragalactic_cmb_tophat`, haven't checked them yet, I'll write here once I do that.

**Comment by keskitalo** [comment]
As far as I recall, the components we discussed were
* galactic foregrounds
* extragalactic foregrounds
* CMB, scalar
* CMB, lensing
* CMB, primordial tensor modes

I am not sure if we really need two different foreground maps at this stage so it should be fine to combine the foregrounds as you seem to do. We _will_ need the primordial tensor modes so that users can set the tensor-to-scalar ratio. Do you think adding the solar system dipole to the scalar fluctuations makes sense?

**Comment by zonca** [comment]
@keskitalo ok, thanks,
about solar system dipole, I have it ready so I can easily add it. I don't have strong preference either way. What do you prefer?

**Comment by keskitalo** [comment]
I prefer not overlooking the dipole. It is very intense and the filtering will undoubtably cause it to leak to higher multipoles.

File: 202002_foregrounds_extragalactic_cmb_tophat/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+* `band` is the channel band
+
+## Combined maps
+
+Also created a single set of maps which is the sum of all components to be used as input to some of the TOD simulations (which don't require running PySM on the fly). They are also reordered to NEST (default for TOAST).
+
+* `combined_foregrounds`: sum of `dust`, `synchrotron`, `freefree`, `ame`, `cib`, `ksz`, `tsz`
+* `cmb_lensing_signal`: `cmb_lensed_solardipole` minus `cmb_unlensed`
+* `cmb_lensed_solardipole_nest`: `cmb_lensed_solardipole`, just reordered to HEALPix NEST
+
```
Comment (zonca) [inline 202002_foregrounds_extragalactic_cmb_tophat/README.md]: @keskitalo here is the plan, I'll add tensor once that is ready.
Do you think some people need only galactic or only extragalactic? should we support that?

Comment (keskitalo) [reply 202002_foregrounds_extragalactic_cmb_tophat/README.md]: I can't think of an application that would only want one and not the other. However, that may well be my lack of imagination.

Comment (zonca) [reply 202002_foregrounds_extragalactic_cmb_tophat/README.md]: ok, let's keep it like this for now

**Comment by zonca** [comment]
@keskitalo I got no answer for tensor modes compatible with Websky, do you have any tensor modes from Planck I can use for now?

**Comment by keskitalo** [comment]
Yes, in the FFP10 folder in the Planck project space.

**Comment by zonca** [comment]
I think `/global/project/projectdirs/planck/data/ffp10/sky/CMB/ffp10_ten_cmb_000_alm.fits` should be it

**Comment by zonca** [comment]
@keskitalo do you know what `tensor-to-scalar ratio` was used in those simulations?

**Comment by jdborrill** [comment]
They should be r=1, with the idea that you then re-scale to your chosen
value of r.

Julian

**Comment by keskitalo** [comment]
From `/global/cfs/cdirs/planck/data/ffp10/sky/CMB/inputs/ffp10_wtensors_params.ini`:
```
tensor_spectral_index(1) = 0
tensor_nrun(1) = 0.0000000000000000
initial_ratio(1) = 0.01
```
So I presume the fiducial value of `r` is `0.01`.

**Comment by jdborrill** [comment]
Ah yes ... it was meant to be 1 to simplify rescaling, but ended up as 0.01
just because.

**Comment by zonca** [comment]
@keskitalo ok, NSIDE 512 is done and validated (took spectra of all maps and compared).

I also prepared the 4 combined components you need for the TOD filtering, see paths in: https://github.com/CMB-S4/s4mapbasedsims/blob/202002_tophat/202002_foregrounds_extragalactic_cmb_tophat/README.md

I'll run 4096 and let you know when it is ready.

**Comment by zonca** [comment]
ok, @keskitalo, 4096 are there as well.
I haven't checked the 4096 maps, if I have time I'll do some checks next week,
but better if you don't wait for that, they should be fine, it is the same codebase of the nside 512.
However please let me know if you find anything wrong.

I'll merge this.

**Comment by zonca** [comment]
now the documentation is available at: https://github.com/CMB-S4/s4mapbasedsims/tree/master/202002_foregrounds_extragalactic_cmb_tophat

**Pull request merged**